### PR TITLE
Update SE to 2.10.03.

### PR DESCRIPTION
**Pull request by kevindecapite**
Refs: https://github.com/aloistr/swisseph/releases/tag/v2.10.03

I copied the files and updated the tests to match the latest results. Most results have only changed by very slight precision values but others have changed a bit more drastically. Are you able to review the items I highlighted to confirm that the new results are indeed correct?

### CODE REVIEW

File: tests/swe_split_deg.phpt [hunk 1]
```diff
@@ -40,7 +40,7 @@ array(5) {
   ["sec"]=>
   int(23)
   ["secfr"]=>
-  float(0)
+  float(23)
```
Comment (kevindecapite) [inline tests/swe_split_deg.phpt:44]: I'm unsure why this value changed from 0 to 23; is this correct?

File: tests/swe_split_deg.phpt [hunk 2]
```diff
@@ -52,7 +52,7 @@ array(5) {
   ["sec"]=>
   int(52)
   ["secfr"]=>
-  float(0)
+  float(52)
```
Comment (kevindecapite) [inline tests/swe_split_deg.phpt:56]: Same question as above.

File: tests/swe_split_deg.phpt [hunk 3]
```diff
@@ -64,7 +64,7 @@ array(5) {
   ["sec"]=>
   int(22)
   ["secfr"]=>
-  float(0)
+  float(22)
```
Comment (kevindecapite) [inline tests/swe_split_deg.phpt:68]: Same question as above.

File: tests/swe_rise_trans_etc.phpt [hunk 4]
```diff
@@ -115,7 +115,7 @@ array(12) {
     ["name"]=>
     string(4) "Moon"
     ["lng"]=>
-    float(221.700155)
+    float(221.700162)
```
Comment (kevindecapite) [inline tests/swe_rise_trans_etc.phpt:118]: The changes in this test appear to be minor precision corrections. Let me know if I'm mistaken.

File: tests/swe_pheno.phpt [hunk 5]
```diff
@@ -58,7 +58,7 @@ array(6) {
   [3]=>
   float(0.519086)
   [4]=>
-  float(-3.987935)
+  float(3.706732)
```
Comment (kevindecapite) [inline tests/swe_pheno.phpt:61]: I know this method was one of the main changes in the SE 2.10.3 release, but I still want to make sure the inversion from negative to positive is correct here?

File: tests/swe_lun_occult.phpt [hunk 6]
```diff
@@ -70,7 +70,7 @@ array(3) {
   ["attr"]=>
   array(8) {
     [0]=>
-    float(86.757454)
+    float(86.757441)
```
Comment (kevindecapite) [inline tests/swe_lun_occult.phpt]: Oddly this test passes in my local build but not in Github. I will fix it later.

File: tests/swe_helio_cross.phpt [hunk 7]
```diff
@@ -28,5 +28,5 @@ array(3) {
   ["rc"]=>
   int(0)
   ["jx"]=>
-  float(2461855.379497)
+  float(2461855.379534)
```
Comment (kevindecapite) [inline tests/swe_helio_cross.phpt:31]: This appears to be just a minor precision adjustment, correct?

File: tests/swe_heliacal_ut.phpt [hunk 8]
```diff
@@ -17,9 +17,9 @@ array(4) {
   ["rc"]=>
   int(0)
   [0]=>
-  float(2452586.417484)
+  float(2452587.410627)
```
Comment (kevindecapite) [inline tests/swe_heliacal_ut.phpt]: These pass locally but not in Github. I will address later.

File: tests/perl_test.phpt [hunk 9]
```diff
@@ -167,7 +167,7 @@ array(3) {
     [1]=>
     float(106.892673)
     [2]=>
-    float(11426.043484)
+    float(11426.043485)
```
Comment (kevindecapite) [inline tests/perl_test.phpt]: This passes locally but not in Github. I will address it later.

**Comment by aloistr** [comment]
This is correct, at elongation 2° the Moon, if it were observable, which it is absolutely not, would be very faint. It is only a theoretical question.

**Comment by aloistr** [comment]
These are all caused by the update of DeltaT for 2021, 2022 etc

swetest0 -bj2459414.104167 -p1 
date (dmy) 18.7.2021 greg.   14:30:00.029 TT      version 2.10.02a
UT:  2459414.103365072     delta t: 69.286625 sec
TT:  2459414.104167000
Epsilon (t/m)     23°26'14.4645   23°26'11.3153
Nutation          -0° 0'15.1441    0° 0' 3.1492
Moon             221°41'19.5158    2°26' 2.8522    0.002465504   14°13'18.6108

swetest -bj2459414.104167 -p1 
date (dmy) 18.7.2021 greg.   14:30:00.029 TT      version 2.10.03
UT:  2459414.103364562     delta t: 69.330609 sec
TT:  2459414.104167000
Epsilon (t/m)     23°26'14.4645   23°26'11.3153
Nutation          -0° 0'15.1441    0° 0' 3.1492
Moon             221°41'19.5158    2°26' 2.8522    0.002465504   14°13'18.6108

**Comment by kevindecapite** [comment]
@aloistr 

Can you explain the reason for this change which now uses the seconds portion of the converted value as the fractional seconds (in `swe_split_deg()`)?

https://github.com/aloistr/swisseph/compare/v2.10.02...v2.10.03?diff=unified#diff-9e215cd6fade77909c85a6ca9891af677c4dada75e1c4331ed673345767d2738R4052

It seems odd to me that when rounding to degrees, minutes or seconds, the fractional seconds value is no longer 0 (zero) but is instead the same as the seconds value itself.

**Comment by kevindecapite** [comment]
FYI: The issue mentioned previously of tests failing in Github while passing locally seems to be related to the M1 ARM chip in my MacBook Air. When using Github-passing values in tests on the ARM, the tests fail. However, using those same values on my Intel Mac passes the tests just fine.

**Pull request merged**